rq/group.py

```diff
@@ -0,0 +1,121 @@
+        """Delete jobs from the group's job registry that have been deleted or expired from Redis.
+        We assume while running this that alive jobs have all been fetched from Redis in fetch_jobs method"""
+
+        pipe = pipeline if pipeline else self.connection.pipeline()
+        job_ids = [as_text(job) for job in list(self.connection.smembers(self.key))]
+        expired_jobs = []
+        with self.connection.pipeline() as p:
+            p.exists(*[Job.key_for(job) for job in job_ids])
+            results = p.execute()
+
+        for i, key_exists in enumerate(results):
+            if not key_exists:
+                expired_jobs.append(job_ids[i])
+
+        if expired_jobs:
+            job_ids = [job for job in job_ids if job not in expired_jobs]
+            pipe.srem(self.key, *expired_jobs)
+
+        if pipeline is None:
+            pipe.execute()
+
+    def enqueue_many(
+        self, queue: Union[str, Queue], job_datas: List['EnqueueData'], pipeline: Optional['Pipeline'] = None
+    ):
+        if isinstance(queue, str):
+            queue = Queue(queue, connection=self.connection)
+        pipe = pipeline if pipeline else self.connection.pipeline()
+
+        jobs = queue.enqueue_many(job_datas, group_id=self.id, pipeline=pipe)
+
+        self._add_jobs(jobs, pipeline=pipe)
+
+        if pipeline is None:
+            pipe.execute()
+
+        return jobs
+
+    def get_jobs(self) -> list:
+        """Retrieve list of job IDs from the group key in Redis"""
+        job_ids = [as_text(job) for job in self.connection.smembers(self.key)]
+        return [job for job in Job.fetch_many(job_ids, self.connection) if job is not None]
+
+    def delete(self):
+        self.connection.delete(self.key)
+
+    def delete_job(self, job_id: str, pipeline: Optional['Pipeline'] = None):
+        pipe = pipeline if pipeline else self.connection.pipeline()
+        pipe.srem(self.key, job_id)
+        if pipeline is None:
+            pipe.execute()
+
+    @classmethod
+    def create(cls, connection: Redis, id: Optional[str] = None):
+        return cls(id=id, connection=connection)
+
+    @classmethod
+    def fetch(cls, id: str, connection: Redis):
+        """Fetch an existing group from Redis"""
+        group = cls(id=id, connection=connection)
+        group.cleanup()
+        if not connection.exists(Group.get_key(group.id)):
+            raise NoSuchGroupError
+        return group
+
+    @classmethod
+    def all(cls, connection: 'Redis') -> List['Group']:
+        "Returns an iterable of all Groupes."
+        group_keys = [as_text(key) for key in connection.smembers(cls.REDIS_BATCH_KEY)]
+        return [Group.fetch(key, connection=connection) for key in group_keys]
+
+    @classmethod
+    def get_key(cls, id: str) -> str:
+        """Return the Redis key of the set containing a group's jobs"""
+        return cls.REDIS_BATCH_NAME_PREFIX + id
+
+    @classmethod
+    def clean_group_registries(cls, connection: 'Redis'):
```

Comment:
I think we can rename this to `Group.clean_registries()`

Reply:
Done